### PR TITLE
identify2 uses the user cache if possible

### PR DESCRIPTION
**Pull request by maxtaco**
- update tests to check that the user cache is being properly accessed

### CODE REVIEW

File: go/engine/identify2_with_uid.go [hunk 1]
```diff
@@ -33,6 +33,112 @@ const (
 	identify2TrackBroke
 )
 
+type identifyUser struct {
+	arg  libkb.LoadUserArg
+	full *libkb.User
+	thin *keybase1.UserPlusAllKeys
+}
+
+func (i *identifyUser) GetUID() keybase1.UID {
```
Comment (patrickxb) [inline go/engine/identify2_with_uid.go]: is there a reason for preferring i.thin over i.full?

Comment (maxtaco) [inline go/engine/identify2_with_uid.go]: It's a good question.  No, but we'd be hosed if they are different!

Comment (maxtaco) [inline go/engine/identify2_with_uid.go]: Happy to switch it if you think it's a good idea.

File: go/engine/identify2_with_uid.go [hunk 2]
```diff
@@ -130,12 +130,12 @@ func (i *identifyUser) isNil() bool {
 func loadIdentifyUser(g *libkb.GlobalContext, arg libkb.LoadUserArg, cache libkb.Identify2Cacher) (*identifyUser, error) {
 	ret := &identifyUser{arg: arg}
 	err := ret.load(g)
+	if ret != nil && ret.full != nil && cache != nil {
```
Comment (patrickxb) [inline go/engine/identify2_with_uid.go]: if ret is nil, won't 132 crash?

Comment (maxtaco) [inline go/engine/identify2_with_uid.go]: yeah, you're right. the real fix was the swap of this check and the reset to `nil` (potentially) right beneath it

**Comment by patrickxb** [comment]
How's performance?

**Comment by maxtaco** [comment]
I couldn't really test performance, so I don't know....

The win of this PR is that after identify the user, we won't need to reload the user again further down the stack. But that might require @mlsteele's PR to go in, which uses the cacheable user load to verify signatures.

**Comment by keybase-ci-visdiff** [comment]
The commits 95c8904e5ac1...0ce76ebbe685 introduce visual changes on linux. <details><summary>:mag_right: 1 changed</summary>

 * Changed: **Paper_Key-Normal** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1480384667861-95c8904e5ac1-0ce76ebbe685-linux/Paper_Key-Normal-95c8904e5ac1.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1480384667861-95c8904e5ac1-0ce76ebbe685-linux/Paper_Key-Normal-0ce76ebbe685.png)  
   Paper_Key-Normal comparison
</summary>

**Pull request merged**